codec-http2/src/test/java/io/netty/handler/codec/http2/HpackDecoderTest.java

```diff
@@ -520,7 +510,126 @@ public void testIncompleteHeaderFieldRepresentation() throws Http2Exception {
         ByteBuf in = Unpooled.wrappedBuffer(input);
         try {
             expectedException.expect(Http2Exception.class);
-            hpackDecoder.decode(0, in, mockHeaders);
+            hpackDecoder.decode(0, in, mockHeaders, true);
+        } finally {
+            in.release();
+        }
+    }
+
+    @Test
+    public void unknownPseudoHeader() throws Exception {
+        ByteBuf in = Unpooled.buffer(200);
+        try {
+            HpackEncoder hpackEncoder = new HpackEncoder(true);
+
+            Http2Headers toEncode = new DefaultHttp2Headers();
+            toEncode.add(":test", "1");
+            hpackEncoder.encodeHeaders(1, in, toEncode, NEVER_SENSITIVE);
+
+            Http2Headers decoded = new DefaultHttp2Headers();
+
+            expectedException.expect(Http2Exception.class);
+            hpackDecoder.decode(1, in, decoded, true);
+        } finally {
+            in.release();
+        }
+    }
+
+    @Test
+    public void disableHeaderValidation() throws Exception {
+        ByteBuf in = Unpooled.buffer(200);
+        try {
+            HpackEncoder hpackEncoder = new HpackEncoder(true);
+
+            Http2Headers toEncode = new DefaultHttp2Headers();
+            toEncode.add(":test", "1");
+            toEncode.add(":status", "200");
+            toEncode.add(":method", "GET");
+            hpackEncoder.encodeHeaders(1, in, toEncode, NEVER_SENSITIVE);
+
+            Http2Headers decoded = new DefaultHttp2Headers();
+
+            hpackDecoder.decode(1, in, decoded, false);
+
+            assertThat(decoded.valueIterator(":test").next().toString(), is("1"));
+            assertThat(decoded.status().toString(), is("200"));
+            assertThat(decoded.method().toString(), is("GET"));
+        } finally {
+            in.release();
+        }
+    }
+
+    @Test
+    public void requestPseudoHeaderInResponse() throws Exception {
+        ByteBuf in = Unpooled.buffer(200);
+        try {
+            HpackEncoder hpackEncoder = new HpackEncoder(true);
+
+            Http2Headers toEncode = new DefaultHttp2Headers();
+            toEncode.add(":status", "200");
+            toEncode.add(":method", "GET");
+            hpackEncoder.encodeHeaders(1, in, toEncode, NEVER_SENSITIVE);
+
+            // decode the headers, we should get an exception, but
+            // the decoded headers object should contain all of the headers
```

Comment:
I still don't completely understand this comment. I guess it is trying to say `:status` should still be contained in `decoded` ... which is why I previously asked if we should validate the header contents after the fact. However if this is not necessary (which I agree it doesn't seem necessary) can we remove/clarify this part of the comment?

Same question in the tests below too.

Reply:
Bad copy/paste, will delete

Reply:
- [x] delete comment